code/main.py

```diff
@@ -1,191 +1,76 @@
-
-    progress.set_status(success_string, False)
-except (RuntimeError, ValueError) as e:
-    wrapped_err = "\n".join(textwrap.wrap(str(e), 70))
-    progress.set_status(f"Error: {wrapped_err}", False)
-    if debug_mode_on:
-        raise
-progress.show_exit_button_and_wait()
+from process_input import process_input
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser('OMCR',
+                                     usage='python cli.py [options] INPUT OUTPUT',
+                                     description='OpenMCR: An accurate and simple exam bubble sheet reading tool.\n'
+                                                 'Reads sheets from input folder, process and saves result in output folder.',
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('input_folder',
+                        help='Path to a folder containing scanned input sheets.\n'
+                             'Sheets with student ID of "9999999999" treated as keys. Ignores subfolders.',
+                        type=Path)
+    parser.add_argument('output_folder',
+                        help='Path to a folder to save result to.',
+                        type=Path)
+    parser.add_argument('--anskeys',
+                        help='Answer Keys CSV file path. If given, will be used over other keys.',
+                        type=Path)
+    parser.add_argument('--formmap',
+                        help='Form Arrangement Map CSV file path. If given, only one answer key may be provided.',
+                        type=Path)
+    parser.add_argument('--variant',
+                        default='75',
+                        choices=['75', '150'],
+                        help='Form variant either 75 questions(default) or 150 questions.')
+    parser.add_argument('-ml', '--multiple',
+                        action='store_true',
+                        help='Convert multiple answers in a question to F, instead of [A|B].')
+    parser.add_argument('-e', '--empty',
+                        action='store_true',
+                        help='Save empty answers as G, blank default.')
+    parser.add_argument('-s', '--sort',
+                        action='store_true',
+                        help="Sort output by students' name.")
+    parser.add_argument('-d', '--debug',
+                        action='store_true',
+                        help='Turn debug mode on. Additional directory with debug data will be created.')
+    parser.add_argument('--mcta',
+                        action='store_true',
+                        help='Output additional files for Multiple Choice Test Analysis.')
+
+    # prints help and exits when called w/o arguments
+    if len(sys.argv) == 1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+
+    args = parser.parse_args()
+
+    image_paths = file_handling.filter_images(file_handling.list_file_paths(args.input_folder))
```

Comment:
I did this because when creating progress tracker for gui you need access to `MainWindow`'s method `create_and_pack_progress` with attribute maximum equal to length of `image_paths`. On the other hand `input_folder` is used only to extract `image_paths`. So i decided to make `process_input` take image paths list instead of input folder path.

Reply:
Oh, I see - that makes sense 👍